### DISCUSSION TITLE
The ability to reach text inside textboxes/documents

### DISCUSSION THREAD

**Discussion by hasantahir** [godbout/Wooshy.docs#131] [General]
Reaching UI elements is fantastic. However, in instances where I am typing something (eg. Word, VS code, this comment in Chrome etc.) I have to steer the mouse (or navigate through the keyboard) back to the desired cursor location. Is it possible through Wooshy to navigate inside the text? Or am I missing something too basic?

For example, I would like Wooshy to take the cursor to the word `where` in the textbox. Presently, Wooshy highlights the whole textbox.

<img width="1084" alt="image" src="https://github.com/godbout/Wooshy.docs/assets/11236148/9ee3de1b-89d8-4f20-ba74-83c6bdf437b0">
 
Thanks, G as always!

**Comment by godbout** [comment]
Hasan my friend!

> Is it possible through Wooshy to navigate inside the text? Or am I missing something too basic?

you're not missing anything. Wooshy uses the macOS Accessibility information to find UI Elements, but this doesn't hold data about where a text is located on a screen. the Accessibility just tell you that in some UI Element, there's some text. i can match this, but i can only get the coordinates of the whole UI Element, not individual texts or words. so this is not possible with Wooshy.

my own Workflow is a combination of Wooshy, and kindaVim. i reach text, and then i use Vim moves to do what i want. see below:

https://github.com/godbout/Wooshy.docs/assets/121373/f52cfc3c-713e-409b-96cd-a178b97a493d

**Reply by godbout** [reply]
not sure what all of your needs are, but for the one you're talking about there, maybe Superkey is more adapted: https://superkey.app
Superkey uses OCR, so it can grab text anywhere on the screen (more or less). even like in PDFs, which Wooshy can't do. but Superkey can't grab icons, buttons without text, etc. at least as far as i know, although it knows supports Accessibility, but i haven't been able to make that work.

**Reply by hasantahir** [reply]
Wow, this is precisely what I wish for. Never quite got the hang of `kindaVim` and also need to watch in slow motion 😃. Do you suggest any resources to get going? Not familiar with Vim at all.

Cheers

**Reply by godbout** [reply]
hmm. have you tried Superkey? not that i don't want to sell my apps 😂️😂️😂️ but kV is **DEFINITELY** for people who are already pretty comfortable with Vim. if you're not familiar with it at all it's probably coz you don't have the need, or it's not your universe. as far as i understand, Superkey would do what you're looking for. but if you really wanna go for a try with kV lemme know again and i'll send all the info to you.

**Reply by hasantahir** [reply]
Thanks G!

I will give Superkey a go. Wooshy has been awesome :-)

**Reply by godbout** [reply]
you're welcome. Superkey and Wooshy may be complementary. or maybe not in your case, maybe Superkey will do the job you're looking for. in any case, enjoy my friend Hasan!
